### ISSUE TITLE
Pyright Complains when Providing Inferred Annotation for `enum.nonmember`

### ISSUE THREAD

**Issue by max-muoto** [microsoft/pyright#7265]
Pyright detects an incorrect type assignment when using the inferred type for an `enum.nonmember` field.

```python
import enum
from typing import reveal_type

class ExampleEnum(enum.StrEnum):
    dict_field = enum.nonmember(
        {"1": "Test", "2": "Test2"}
    )

reveal_type(ExampleEnum.dict_field) # Type of "ExampleEnum.dict_field" is "dict[str, str]
```

[Pyright Playground](https://pyright-play.net/?strict=true&code=FDCWFsAcHsCcBcAEBTAdgV3MAZra5F4BPSUVAc0QhgUVmQDdkBDAGwH1jJkQBjV5gGdBiAKIAPZlFbJRGcAAo0mAHQBleLDmYAlAC5giI4gAmoXvHbZQyVicQBeFPJWpoqcMnAAjZLAWGxkEA3gBEAIyheoihACrIgvChADQxAExRMfGJGQC%2BgUY6IPRMbJwkyAoSUpAy2uAqZhZWNnZFQA).

Manually providing this annotation:

```python
import enum
from typing import reveal_type

class ExampleEnum(enum.StrEnum):
    dict_field: dict[str, str] = enum.nonmember(
        {"1": "Test", "2": "Test2"} 
    )

reveal_type(ExampleEnum.dict_field)
```

[Pyright Playground](https://pyright-play.net/?strict=true&code=FDCWFsAcHsCcBcAEBTAdgV3MAZra5F4BPSUVAc0QhgUVmQDdkBDAGwH1jJkQBjV5gGdBiAKIAPZlFbJRGcAAo0mAHQBleLDmYAlAC5giI4gAmoXvHbZQyVib2nz8ANqDNAGkRvYAXUQBeFHkVVGhUcGRwACNkWAVDY0SAbwAiAEYUhxSAFWQ3FM8UgCZMxBy8%2BBKAXwSjHRB6JjZOEmQFCSlIGW1wFTMLKxs7eqA
).

You'll get: `Expression of type "nonmember[dict[str, str]]" cannot be assigned to declared type "Literal[ExampleEnum.dict_field]"   "nonmember[dict[str, str]]" cannot be assigned to type "Literal[ExampleEnum.dict_field]"`

Python Version: Python 3.12
Pyright Version: 1.1350

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by max-muoto** [comment]
Great, thanks Eric!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.351, which I just published.
